### ISSUE TITLE
PubQuizzer improvement site.

### ISSUE THREAD

**Issue by holden** [razrfly/quizfindr#5]
@holden commented on [Fri Jul 07 2017](https://github.com/razrfly/questionone/issues/2718)

# Overview

Our own version of: http://www.pubquizzers.com/

* Verified quizzes
* Unverified quizzes
* Dates of last verified
* Source of information (eg the Pub, QuestionOne, PubQuizzers.com, or Geeks)

# More

* [ ] Create an API for our own Q1 quizzes.
* [ ] Create simple API for PubQuizzers & GeekWhoDrink to regularly import quizzes / keep them up-to-date, unverified, credit the source.

# Quizzes

* Venue name
* Address
* Postcode
* GeoCoordinates
* City
* Country
* Jackpot
* Entry fee
* Date/Frequency of quiz
* Website (maybe 2, venue website, quiz info page?, facebook, etc.)
* Source
* QM name (link to website?)

Then make a series of daily rake tasks to import data into the site.

The basic idea is based around verified data.

* Looking for new Quizzes vs Quizzes already in our DB.
* Verified vs Unverified...
* Q1 data & GWD data is always 100% verified, because we can trust it.
* PubQuizzer data is by default unverified.
* But once someone logs in and somehow proves it is their Quiz (or they show us a website verifying the quiz) we make it verified and ignore the PubQuizzer data.

# Users

* Devise (username, first_name, last_name)
* Simple STI types (Player / Venue / QuizMaster)?

# Blogs

Verification process, need a fair process for a quiz to get regularly verified, must be regularly maintained, eg automated emails goes out once per 3-6 months and they must somehow provide proof of life.

# Forum (eventually)

Using https://github.com/thredded/thredded
or possibly: https://github.com/discourse/discourse

But I like the idea of thredded better as we can full integrate it and it's simpler for very basic forum / comments system.

# Ratings

Simple rating system.

**Comment by tchorzewski1991** [comment]
Referring to the models ... I look on the description, that you made on the issue, and I am wondering if will it be good idea to extract address as separate model?

So my idea for the models is:

**Quiz:**
- Quiz name
- Source
- Jackpot
- Entry fee
- Currency?
- Date
-Website

**HAS_ONE:**

**Address or Location**
- Address
- City
- Country
- State
- Postcode
- Geo's

Smaller Quiz table, little more SR principle, more lightweight models (for example geocoder deps
will go to Location/Address model)

@holden please give some review on that idea. If it's not good, I will made single model

**Comment by holden** [comment]
No, it would make working with the geocoder gem difficult.  Would need to make extra methods just to do simple things like.:

`Venue.near('London, NW3', 20)`

vs

```
addresses = Address.near('London, NW3', 20)
addresses.map(&:quiz)
```

Not worth it. ;-(

**Comment by tchorzewski1991** [comment]
Good point! 🤔 👍

**Comment by holden** [comment]
Plus this table won't ever be that big.
